src/openforms/registrations/contrib/objects_api/json_schema.py

```diff
@@ -168,3 +168,39 @@ def get_missing_required_paths(
         missing_paths.append(r_path.segments)
 
     return missing_paths
+
+
+def json_schema_matches(
+    *, variable_schema: ObjectSchema, target_schema: ObjectSchema
+) -> bool:
+    """Return whether the deduced JSON Schema of a variable is compatible with the target object (sub) JSON Schema.
+
+    In other terms, this determines whether the variable can be mapped to a specific location. Currently,
+    only a limited subset of features is supported. For instance, ``format`` constraints are supported
+    if the type is a string, however no inspection is done on ``properties`` if it is an object.
+    """
+    if "type" not in target_schema:
+        return True
+
+    if "type" not in variable_schema:
+        # 'type' is in target but not in variable
+        return False
+
+    target_types: str | list[str] = target_schema["type"]
+    if not isinstance(target_types, list):
+        target_types = [target_types]
+
+    variable_types: str | list[str] = variable_schema["type"]
+    if not isinstance(variable_types, list):
+        variable_types = [variable_types]
+
+    if not set(variable_types).issubset(target_types):
+        return False
+
+    if "string" in target_types and (target_format := target_schema.get("format")):
```

Comment:
Would a case be possible where the `variable_format` is not `None` but the `target_format` is? In this case I think the `json_schema_matches` function would return `True`, but I am not sure if it is intended?

Reply:
I added a test covering this behavior. It is hard to think about all the possible cases :sweat_smile: